File: src/main/java/com/silverboxsoft/dynamodbtool/dao/PutItemDao.java
package com.silverboxsoft.dynamodbtool.dao;

import java.net.URISyntaxException;
import java.util.Map;

import com.silverboxsoft.dynamodbtool.classes.DynamoDbConnectInfo;

import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
import software.amazon.awssdk.services.dynamodb.model.PutItemResponse;
import software.amazon.awssdk.services.dynamodb.model.TableDescription;

/**
 * https://github.com/awsdocs/aws-doc-sdk-examples/blob/master/javav2/example_code/dynamodb/src/main/java/com/example/dynamodb/PutItem.java
 * 
 * @author tanakaeiji
 *
 */
public class PutItemDao extends AbsDao {

	public PutItemDao(DynamoDbConnectInfo connInfo) {
		super(connInfo);
	}

	public PutItemResponse putItem(TableDescription tableInfo, Map<String, AttributeValue> dynamoDbRec)
			throws URISyntaxException {
		String tableName = tableInfo.tableName();
		DynamoDbClient ddb = getDbClient();
		try {
			PutItemRequest request = PutItemRequest.builder().tableName(tableName).item(dynamoDbRec).build();
			return ddb.putItem(request);
		} finally {
			ddb.close();
		}
	}
}

File: src/main/java/com/silverboxsoft/dynamodbtool/dao/PutItemDao.kt
package com.silverboxsoft.dynamodbtool.dao

import kotlin.Throws
import com.silverboxsoft.dynamodbtool.classes.DynamoDbConnectInfo
import java.net.URISyntaxException
import software.amazon.awssdk.services.dynamodb.model.*

/**
 * https://github.com/awsdocs/aws-doc-sdk-examples/blob/master/javav2/example_code/dynamodb/src/main/java/com/example/dynamodb/PutItem.java
 *
 * @author tanakaeiji
 */
class PutItemDao(connInfo: DynamoDbConnectInfo) : AbsDao(connInfo) {
    @Throws(URISyntaxException::class)
    fun putItem(tableInfo: TableDescription?, dynamoDbRec: Map<String, AttributeValue>): PutItemResponse {
        val tableName = tableInfo!!.tableName()
        val ddb = dbClient
        return try {
            val request = PutItemRequest.builder().tableName(tableName).item(dynamoDbRec).build()
            ddb.putItem(request)
        } finally {
            ddb.close()
        }
    }
}